File: app/src/main/java/com/se/music/fragment/QuickControlsFragment.java
package com.se.music.fragment;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.facebook.drawee.view.SimpleDraweeView;
import com.se.music.R;
import com.se.music.activity.PlayMusicActivity;
import com.se.music.service.MusicPlayer;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

/**
 * =======================================================
 * 作者：gaojin
 * 日期：2017/3/3 下午4:47
 * 描述：底部音乐快捷控制栏
 * 备注：备注：Copyright © 2010-2017. gaojin All rights reserved.
 * =======================================================
 */
public class QuickControlsFragment extends BaseFragment {

    @BindView(R.id.playbar_img)
    SimpleDraweeView mAlbum;

    @BindView(R.id.rl_layout)
    RelativeLayout relativeLayout = null;

    @BindView(R.id.playbar_info)
    TextView mPlaybarInfo;

    @BindView(R.id.playbar_singer)
    TextView mPlaybarSinger;

    @BindView(R.id.play_list)
    ImageView mPlayList;

    @BindView(R.id.control)
    ImageView mControl;

    @BindView(R.id.play_next)
    ImageView mPlayNext;

    public static QuickControlsFragment newInstance() {
        return new QuickControlsFragment();
    }

    @OnClick(R.id.rl_layout)
    void rlLayout() {
        Intent intent = new Intent(getActivity(), PlayMusicActivity.class);
        startActivity(intent);
        getActivity().overridePendingTransition(R.anim.push_down_in, R.anim.push_up_out);
    }

    @OnClick(R.id.play_list)
    void playList() {
    }

    @OnClick(R.id.control)
    void control() {
        MusicPlayer.Companion.playOrPause();
    }

    @OnClick(R.id.play_next)
    void playNext() {
        MusicPlayer.Companion.nextPlay();
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.view_quick_controls, container, false);
        ButterKnife.bind(this, rootView);
        return rootView;
    }

    @Override
    public void onResume() {
        super.onResume();
        updateFragment();
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
    }

    public void updateFragment() {
        mPlaybarInfo.setText(MusicPlayer.Companion.getTrackName());
        mPlaybarSinger.setText(MusicPlayer.Companion.getArtistName());
        if (MusicPlayer.Companion.getIsPlaying()) {
            mControl.setImageResource(R.drawable.playbar_btn_pause);
        } else {
            mControl.setImageResource(R.drawable.playbar_btn_play);
        }
        if (MusicPlayer.Companion.getAlbumPic() != null) {
            mAlbum.setImageURI(MusicPlayer.Companion.getAlbumPic());
        }
    }

    @Override
    public void updatePlayInfo() {
        updateFragment();
    }
}

File: app/src/main/kotlin/com/se/music/fragment/QuickControlsFragment.kt
package com.se.music.fragment

import android.animation.ObjectAnimator
import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.view.animation.LinearInterpolator
import android.widget.ImageView
import android.widget.TextView
import com.se.music.R
import com.se.music.activity.PlayingActivity
import com.se.music.base.BaseFragment
import com.se.music.service.MusicPlayer
import com.se.music.utils.getLargeImageUrl
import com.se.music.utils.isContentEmpty
import com.se.music.utils.loadUrl


/**
 *Author: gaojin
 *Time: 2018/5/23 下午10:32
 */

class QuickControlsFragment : BaseFragment(), View.OnClickListener {

    companion object {
        fun newInstance(): QuickControlsFragment {
            return QuickControlsFragment()
        }
    }

    private lateinit var album: ImageView
    private lateinit var playBarSongName: TextView
    private lateinit var playBarSinger: TextView
    private lateinit var playList: ImageView
    private lateinit var control: ImageView
    private lateinit var playNext: ImageView
    private lateinit var mLogan: TextView
    private lateinit var circleAnim: ObjectAnimator

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
        val root = inflater.inflate(R.layout.view_quick_controls, container, false)
        album = root.findViewById(R.id.play_bar_img)
        playBarSongName = root.findViewById(R.id.play_bar_song_name)
        playBarSinger = root.findViewById(R.id.play_bar_singer)
        playList = root.findViewById(R.id.play_list)
        control = root.findViewById(R.id.control)
        playNext = root.findViewById(R.id.play_next)
        mLogan = root.findViewById(R.id.music_logan)

        val albumCenterPoint = resources.getDimension(R.dimen.bottom_fragment_album_size) / 2
        album.pivotX = albumCenterPoint
        album.pivotY = albumCenterPoint

        circleAnim = ObjectAnimator.ofFloat(album, "rotation", 0f, 360f).apply {
            interpolator = LinearInterpolator()
            repeatCount = -1
            duration = 12000
            start()
        }

        root.setOnClickListener(this)
        root.elevation = 100f

        playList.setOnClickListener(this)
        control.setOnClickListener(this)
        playNext.setOnClickListener(this)
        return root
    }

    override fun onResume() {
        super.onResume()
        updateFragment()
    }

    override fun onClick(v: View) {
        when {
            v.id == R.id.play_list -> {
            }
            v.id == R.id.control -> MusicPlayer.playOrPause()
            v.id == R.id.play_next -> MusicPlayer.nextPlay()
            else -> {
                val intent = Intent(activity, PlayingActivity::class.java)
                startActivity(intent)
                activity!!.overridePendingTransition(R.anim.push_down_in, R.anim.push_up_out)
            }
        }
    }

    override fun updatePlayInfo() {
        updateFragment()
    }

    private fun updateFragment() {

        album.loadUrl(MusicPlayer.getAlbumPic().getLargeImageUrl(), R.drawable.player_albumcover_default)

        if (MusicPlayer.getTrackName().isContentEmpty() && MusicPlayer.getArtistName().isContentEmpty()) {
            playBarSongName.visibility = View.GONE
            playBarSinger.visibility = View.GONE
            album.visibility = View.GONE
            mLogan.visibility = View.VISIBLE
        } else {
            playBarSongName.visibility = View.VISIBLE
            playBarSinger.visibility = View.VISIBLE
            album.visibility = View.VISIBLE
            mLogan.visibility = View.GONE
            playBarSongName.text = MusicPlayer.getTrackName()
            playBarSinger.text = MusicPlayer.getArtistName()
        }

        if (MusicPlayer.isPlaying()) {
            control.setImageResource(R.drawable.playbar_btn_pause)
            circleAnim.resume()
        } else {
            control.setImageResource(R.drawable.playbar_btn_play)
            circleAnim.pause()
        }
    }
}